pkg/skaffold/build/model.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package build
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+)
+
+// artifactChanModel models the artifact dependency graph using a set of channels.
+// Each artifact has a status struct that has success and a failure channel which it closes once it completes building by calling either markSuccess or markFailure respectively.
+// This notifies all listeners waiting for this artifact of a successful or failed build.
+// Additionally it has a reference to the channels for each of its dependencies.
+// Calling `waitForDependencies` ensures that all required artifacts' channels have already been closed and as such have finished building before the current artifact build starts.
+type artifactChanModel struct {
+	artifact                 *latest.Artifact
+	artifactStatus           status
+	requiredArtifactStatuses []status
+	concurrencySem           chan bool
```

Comment:
This describe what it is, not so much what it's for.  `buildToken`?

Reply:
renamed